examples/workflows/upstream/volumes_emptydir.py

```diff
@@ -0,0 +1,23 @@
+from hera.workflows import (
+    Container,
+    Workflow,
+    models as m,
+)
+
+with Workflow(
+    generate_name="volumes-emptydir-",
+    entrypoint="volumes-emptydir-example",
+    volumes=[m.Volume(name="workdir", empty_dir=m.EmptyDirVolumeSource())],
```

Comment:
@elliotgunton we [already have a volume mixin](https://github.com/argoproj-labs/hera-workflows/pull/437/files#diff-079dae561586ca661f24348fd6560e265e3c6eba6650abd05491201598ae9288R233)! [Here's a usage pattern on container](https://github.com/argoproj-labs/hera-workflows/pull/437/files#diff-8ee20f78bd3bb4a719ab88b4388313de94453566413efd6a7d1d5927a8b2b770R28) and [how it's built](https://github.com/argoproj-labs/hera-workflows/pull/437/files#diff-8ee20f78bd3bb4a719ab88b4388313de94453566413efd6a7d1d5927a8b2b770R94). I think one thing that's missing at the moment, on workflow, is the automatic creation of PVCs for any volumes that are created i.e. if I set a `Volume(size=5Gi, mount_path=/mnt)` on a Step/Task template, workflow should perform a `.to_pvc` on the volume, on the user behalf